### ISSUE TITLE
721619e8890e ("core: Parse all memory DT nodes") has broken master on QEMU 

### ISSUE THREAD

**Issue by jenswi-linaro** [OP-TEE/optee_os#4093]
It seems that 721619e8890e ("core: Parse all memory DT nodes") has broken master on QEMU.

```
D/TC:0 0 init_canaries:191 #Stack canaries for stack_thread[1] with top at 0xe1a37f8
D/TC:0 0 init_canaries:191 watch *0xe1a37fc
D/TC:0 0 init_canaries:191 #Stack canaries for stack_thread[2] with top at 0xe1a5838
D/TC:0 0 init_canaries:191 watch *0xe1a583c
I/TC:0 0 init_external_dt:1056 Non-secure external DT found
E/TC:0 0 check_phys_mem_is_outside:335 Non-sec mem (0xe000000:0x100000) overlaps map (type 14 0xe000000:0x100000)
E/TC:0 0 Panic at core/arch/arm/mm/core_mmu.c:336 <check_phys_mem_is_outside>
E/TC:0 0 TEE load address @ 0xe100000
E/TC:0 0 Call stack:
E/TC:0 0  0x0e11aee5
E/TC:0 0  0x0e12e18d
E/TC:0 0  0x0e11bb7d
E/TC:0 0  0x0e11bd21
E/TC:0 0  0x0e11aaed
E/TC:0 0  0x0e11ac67
E/TC:0 0  0x0e11ada9
E/TC:0 0  0x0e100174
```

Anything I'm missing?

**Comment by jforissier** [comment]
Oh sh* you're right... I'm quite angry about that :angry: because I pointed out the issue to @marex in #4044. Then I saw that CI was OK therefore I believed QEMU had been fixed. But in fact the status was green only because IBART was not run :cry: 

Unless someone proposes a proper fix, I'm going to revert 721619e.

**Comment by Emantor** [comment]
FWIW, OP-TEE assumes that the DT memory node only contains non-secure world memory, while the original memory node DT specification contains "…all physical memory". For platforms with the generic RAM layout we remove the secure world memory from the parsed DT memory node. However since for QEMU there is no generic RAM layout and everything is parsed from DT, no removal is done.

**Comment by jforissier** [comment]
@Emantor so by this logic, should we `carve_out_phys_mem()` all areas declared secure-only via `register_phys_mem*()`?

**Comment by Emantor** [comment]
@jforissier IMO the notion of checking that the physical memory does not contain secure memory is flawed, and we should rather ensure that all of the secure memory is covered by reserved-memory nodes. I'd be in favor of removing the `check_phys_mem_is_outside()` function.

**Comment by Emantor** [comment]
> ...and replace it with some new function `check_phys_mem_is_reserved()` instead? Did I understand correctly?
> If so I can try to write a patch.

Since we generate the reserved memory nodes from the same information we used to check the memory can't we simply remove the function? IDK if there are cases where we don't insert the nodes.

**Comment by jforissier** [comment]
Tentative fix in #4096. Checking that OP-TEE private memory is either not in the non-secure memory nodes, or is declared as reserved-memory is a separate thing that could be done later IMO.

**Comment by jbech-linaro** [comment]
> But in fact the status was green only because IBART was not run

Once again a certificate issue. I thought I had sorted it last time this happened but apparently not. The"certbot" does renew the certificates. However, one need to re-run/ nginx to start using the new certificates. I did put a cronjob a couple of months ago that should restart the nginx every night at 2 a'clock (UTC+2). But it looks like this hasn't happened, alternative there is some other issue related with it. Doing some re-runs of pending jobs and checking various things with it.
